framework/src/modules/dpos_v2/commands/pom.ts

```diff
@@ -143,33 +159,49 @@ export class ReportDelegateMisbehaviorCommand extends BaseCommand {
 			getMethodContext(),
 			header1.generatorAddress,
 		);
-		/*
-			Check block signatures validity
-		*/
 		header1.validateSignature(generatorKey, chainID);
 		header2.validateSignature(generatorKey, chainID);
 
-		/*
-			Update sender account
-		*/
 		const delegateAccountBalance = await this._tokenMethod.getAvailableBalance(
 			getMethodContext(),
 			delegateAddress,
 			this._tokenIDDPoS,
 		);
 
-		const reward =
-			REPORTING_PUNISHMENT_REWARD > delegateAccountBalance
-				? delegateAccountBalance
-				: REPORTING_PUNISHMENT_REWARD;
-
 		delegateAccount.pomHeights.push(currentHeight);
 
-		if (delegateAccount.pomHeights.length >= MAX_POM_HEIGHTS) {
+		this.events.get(DelegatePunishedEvent).log(context, {
+			address: delegateAddress,
+			height: currentHeight,
+		});
+
+		if (delegateAccount.pomHeights.length === POM_LIMIT_BANNED) {
 			delegateAccount.isBanned = true;
+
+			this.events.get(DelegateBannedEvent).log(context, {
+				address: delegateAddress,
+				height: currentHeight,
+			});
 		}
+
+		const currentWeight = getDelegateWeight(
+			BigInt(this._config.factorSelfVotes),
+			delegateAccount.selfVotes,
+			delegateAccount.totalVotesReceived,
+		);
+
+		const eligibleDelegateStore = this.stores.get(EligibleDelegatesStore);
+		await eligibleDelegateStore.update(context, delegateAddress, currentWeight, delegateAccount);
+
 		await delegateSubStore.set(context, delegateAddress, delegateAccount);
 
+		/* Checking if the reward is greater than the delegate's balance. If it is, then it will set the
+		reward to the delegate's balance. */
+		const reward =
+			REPORTING_PUNISHMENT_REWARD > delegateAccountBalance
+				? delegateAccountBalance
+				: REPORTING_PUNISHMENT_REWARD;
```

Comment:
IMHO, the comment here is not needed, as the code is super simple and obvious 🙂

The expression is OK, but if I was writing it, I would go for the less complex version:

```typescript
const reward = Math.min(REPORTING_PUNISHMENT_REWARD, delegateAccountBalance)
```

Reply:
As this is a draft PR, comments are ok and will be cleaned up in the final draft.

As for the suggested implementation,  looks good 😉

Reply:
Ahhh good to know 💡